src/Core/Mews.Fiscalizations.Core/Model/Legal/EuropeanUnionTaxpayerIdentificationNumber.cs

```diff
@@ -25,7 +25,7 @@ private EuropeanUnionTaxpayerIdentificationNumber(EuropeanUnionCountry country,
                     t => country.TaxpayerNumberPattern,
                     f => country.TaxpayerNumberPatternWithoutCountryCodePrefix
                 );
-                return StringValidations.RegexMatch(taxpayerNumber, pattern).Map(n => new EuropeanUnionTaxpayerIdentificationNumber(c, n));
+                return StringValidations.RegexMatch(taxpayerNumber.Trim(), pattern).Map(n => new EuropeanUnionTaxpayerIdentificationNumber(c, n));
```

Comment:
I don't like the trimming. Is it necessary? We're validating stuff but this doesn't cause error, it just corrects it instead.

Reply:
I believe it should be completely ok to fill the tax id as CZ 1234... and not CZ1234.....

or its possible that someone made amistake and added an extra space at the end (which happens A LOT), so correcting these cases should be fine IMO.

this would still fail if you only insert whitespaces anyway.

Reply:
As discussed on call, it's inconsistent - we validate nulls but then trim. So we should either just ignore nulls and replace with empty string or validate both or something. 

We're moving the trim to the call site.